hphp/hack/src/parser/rust_parser_errors.rs

```diff
@@ -4196,6 +4196,20 @@ where
                 }
             }
 
+            if Self::list_contains_predicate(|x| x.is_xhp(), &cd.classish_modifiers) {
+                match Self::token_kind(&cd.classish_keyword) {
+                    Some(TokenKind::Interface) => self.errors.push(Self::make_error_from_node(
+                        node,
+                        errors::invalid_xhp_classish("Interfaces"),
+                    )),
+                    Some(TokenKind::Trait) => self.errors.push(Self::make_error_from_node(
+                        node,
+                        errors::invalid_xhp_classish("Traits"),
+                    )),
+                    _ => (),
```

Comment:
Does this need to handle `xhp enum Foo {}`?

Reply:
Fixed! Good catch